docs/tutorial-01.md

```diff
@@ -0,0 +1,210 @@
+
+First step is to install buildah:
+
+    dnf -y install buildah
+
+After installing buildah we can see there are no images installed. The `buildah images` command will list all the images:
+
+    buildah images
+
+We can also see that there are also no containers by running:
+
+    buildah containers
+  
+When you build a working container from an existing image, buildah defaults to appending '-working-container' to the image's name to construct a name for the container. The buildah CLI conveniently returns the name of the new container. You can take advantage of this by assigning the returned value to a shell varible using standard shell assignment :
+
+    container=$(buildah from fedora)
+
+Of course you can always just run `build from fedora` and then list the containers to see the contianer name. As mentioned previously it will be fedora-working-container in this case.
+
+See the name:  
+
+    echo $container
+
+What can we do with this new container? Let's try running bash:
+
+    buildah run $container bash
+    
+Notice we get a new shell prompt because we running a bash shell inside of the container. It should be noted that `buildah run` is not intended for running production containers. It is for helping debug during the build process. 
+
+Let's try running something else:
+
+    buildah run $container java
+
+Oops. Java is not installed. A message containing something like the following was returned.
+
+    container_linux.go:274: starting container process caused "exec: \"java\": executable file not found in $PATH"
+    
+Lets try installing it using:
+    
+    buildah run $container -- dnf install java
+
+The `--` syntax basically says: no more `buildah run` command options after this point. It is required if the command we specify includes command line options which are not meant for buildah. 
+
+Now running `java` will show that Java has been installed. It will return the `Usage`:
+
+## Building a container from scratch
+
+One of the advantages of using `buildah` to build OCI compliant container images is that you can easily build a container image from scratch and therefore exclude unnecessary packages from your image. E.g. most final container images for production probably don't need a package manager like `dnf`. 
+
+Let's build a container from scratch. The special "image" name scratch tells buildah to create an empty container:
+
+    newcontainer=$(buildah from scratch)
+  
+You can see this new empty container by running:
+
+    buildah containers
+  
+Its container name is working-container by default. And it's stored in the `$newcontainer` variable. Notice the image name is scratch. And when we run:
+
+    buildah images
+  
+We don't see it listed. That is because "scratch" is not an image. It is an empty container.
```

Comment:
"scratch" won't show up in the containers list, either, and commands that deal with containers won't see it.  It's just a name that we specifically recognize as meaning "no image".  I'm harping on this, probably a little too much, because I think it's very important that we are always clear about whether we're talking about containers or images, so that readers don't get confused.

Reply:
scratch does show up. as the IMAGE NAME of working-container.

ac8fa6be0f0a     *                  scratch                          working-container

Reply:
Does this help?

Its container name is working-container by default. And it's stored in the `$newcontainer` variable. Notice the image name (IMAGE NAME) is "scratch". This just indicates that there is no real image yet. i.e. It is containers/storage but there is no representation in containers/image. So when we run:

    buildah images
  
We don't see the image listed. That is because "scratch" is not an image. It is an empty container.

Reply:
Nope. better:
We don't see the image listed. There is no image. It is an empty container.

Reply:
Ah, the "it" is what confused me, because the previous sentence was talking about images.  Maybe "The container has not contents."?